miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/StorageLocationRestController.java

```diff
@@ -0,0 +1,200 @@
+import uk.ac.bbsrc.tgac.miso.core.data.impl.StorageLocation.LocationUnit;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.StorageLocationDto;
+import uk.ac.bbsrc.tgac.miso.service.StorageLocationService;
+
+@Controller
+@RequestMapping("/rest/storagelocations")
+public class StorageLocationRestController extends RestController {
+
+  @Autowired
+  private StorageLocationService storageLocationService;
+
+  @RequestMapping(value = "/freezers", method = RequestMethod.GET)
+  public @ResponseBody List<StorageLocationDto> getFreezers() {
+    return storageLocationService.listFreezers().stream()
+        .map(freezer -> Dtos.asDto(freezer, false, false))
+        .collect(Collectors.toList());
+  }
+
+  @RequestMapping(value = "/bybarcode", method = RequestMethod.GET)
+  public @ResponseBody StorageLocationDto getLocation(@RequestParam(name = "q", required = true) String barcode) {
+    StorageLocation location = storageLocationService.getByBarcode(barcode);
+    if (location == null) {
+      throw new RestException("storage location not found", Status.NOT_FOUND);
+    }
+    return Dtos.asDto(location, true, false);
+  }
+
+  @RequestMapping(value = "/{id}/children", method = RequestMethod.GET)
+  public @ResponseBody List<StorageLocationDto> getChildLocations(@PathVariable(name = "id", required = true) long id) {
+    StorageLocation location = storageLocationService.get(id);
+    if (location == null) {
+      throw new RestException("storage location not found", Status.NOT_FOUND);
+    }
+    return location.getChildLocations().stream()
+        .map(child -> Dtos.asDto(child, false, false))
+        .collect(Collectors.toList());
+  }
+
+  @RequestMapping(value = "/freezers", method = RequestMethod.POST)
+  @ResponseStatus(HttpStatus.CREATED)
+  public @ResponseBody StorageLocationDto createFreezer(@RequestBody StorageLocationDto dto) {
+    return doSave(dto);
+  }
+
+  @RequestMapping(value = "/freezers/{id}", method = RequestMethod.PUT)
+  public @ResponseBody StorageLocationDto update(@PathVariable(name = "id", required = true) long id, @RequestBody StorageLocationDto dto)
+      throws IOException {
+    if (dto.getId() != id) {
+      throw new RestException("Location ID mismatch", Status.BAD_REQUEST);
+    }
+    getFreezer(id); // checks existing freezer
+    return doSave(dto);
+  }
+
+  private StorageLocationDto doSave(StorageLocationDto dto) {
+    StorageLocation freezer = Dtos.to(dto);
+    long savedId = storageLocationService.saveFreezer(freezer);
+    StorageLocation saved = storageLocationService.get(savedId);
+    return Dtos.asDto(saved, true, true);
+  }
+
+  @RequestMapping(value = "/freezers/{id}/shelves", method = RequestMethod.POST)
+  public @ResponseBody StorageLocationDto addFreezerShelf(@PathVariable(name = "id", required = true) long id) {
+    StorageLocation freezer = getFreezer(id);
+    int lastShelf = freezer.getChildLocations().stream().filter(loc -> {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove useless curly braces around statement and then remove useless return keyword [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1602)